packages/react-dom/src/client/ReactDOMFiberComponent.js

```diff
@@ -866,31 +867,35 @@ export function diffHydratedProperties(
       break;
     case 'input':
       ReactDOMFiberInput.initWrapperState(domElement, rawProps);
+      props = ReactDOMFiberInput.getHostProps(domElement, rawProps);
       trapBubbledEvent(TOP_INVALID, domElement);
       // For controlled components we always need to ensure we're listening
       // to onChange. Even if there is no listener.
       ensureListeningTo(rootContainerElement, 'onChange');
       break;
     case 'option':
       ReactDOMFiberOption.validateProps(domElement, rawProps);
+      props = ReactDOMFiberOption.getHostProps(domElement, rawProps);
       break;
     case 'select':
       ReactDOMFiberSelect.initWrapperState(domElement, rawProps);
+      props = ReactDOMFiberSelect.getHostProps(domElement, rawProps);
       trapBubbledEvent(TOP_INVALID, domElement);
       // For controlled components we always need to ensure we're listening
       // to onChange. Even if there is no listener.
       ensureListeningTo(rootContainerElement, 'onChange');
       break;
     case 'textarea':
       ReactDOMFiberTextarea.initWrapperState(domElement, rawProps);
+      props = ReactDOMFiberTextarea.getHostProps(domElement, rawProps);
```

Comment:
With this addition, I think this set of operations is exactly the same for hydration and client-rendering. Also I think they need to be the same for SSR to align warning behavior. Maybe these should be in a shared function.